website/content/projects/2019/spec.md

```diff
@@ -118,11 +118,62 @@ The project will span seven weeks (Weeks 4 - 10), with a final review and team p
+
+So, the WDL team can initially focus on this:
+
+- Writing a draft specification of the WDL language
+- Writing a module for loading WDL files. This module must do some basic validation of the file (ensuring it meets the specification of the WDL language)
+
+After this, you will find you will need to work more closely with the Game State team to figure out how to load the contents of the WDL file into the data structures provided by Game State.
+
+
+### Game State
+
+This team will be responsible for designing the data structures that will store the state of the game in memory, as well as a module to manipulate those data structures. A good initial goal is to focus on *designing* these modules by writing the modules' header files without yet writing the implementation of the functions promised by the module's interface. This will get you thinking about what data structures to use, how they will relate to other data structures, what operations you want to allow on each data structure.
+
+
+### Action Management
+
+The Action Management team is responsible for deciding what actions are possible, and how to model the conditions on an action as well as the effects of an action. Our sample WDL files provide examples of how these action conditions/effects could be specified in the WDL file, but that information will then have to be stored in memory in a general-purpose way that makes it easy to process actions during the game.
+
+As such, this team can initially focus on the following:
+
+- Writing a draft specification of what information is necessary to specify an action (you will then be able to use this to convey to the WDL team your exact requirements for specifying actions in the WDL file)
+- Designing, but not yet implementing, a module with the data structures that would allow you to model actions and their processing.
+
+
+### Checkpointing
+
+This component will be responsible for saving and restoring game state data and, as such, the implementation of this component will be tightly coupled to the design decisions made by the Game State team and the Action Management team. However, there are two broad approaches we could follow: dumping the entire game state to a file, or dumping only a log of actions that have taken place since loading the WDL file. 
+
+So, to get started, it can be useful to start looking into how to save this kind of information into a binary file, possibly using existing file formats. Writing two sample programs, one that saves information from a mock game (not specifically tied to the WDL or Game State we will eventually use), and another that reads that information and prints it out, can be a good way to start figuring out the mechanics of checkpointing, which will make it easier to accommodate the requirements coming from WDL, Game State, and Action Management later on.
```

Comment:
`s/file formats/serialization libraries/`? "file formats" seems like we're suggesting that they use another game's save file format, which wouldn't be logical.

Reply:
Pushed a change to this.